src/Mafia/Script.hs

```diff
@@ -136,7 +136,7 @@ parseScript path source =
 
 renderSubmoduleUrl :: Submodule -> Text
 renderSubmoduleUrl s =
-  "git@github.com:" <> submoduleUser s <> "/" <> submoduleRepo s
+  "https://github.com/" <> submoduleUser s <> "/" <> submoduleRepo s
```

Comment:
This is really a breaking change for mafia scripts, can the protocol be added to the Pragma in a compatible way? 

e.g.

```
data Submodule =
  Submodule {
      submoduleProtocol :: !Protocol
    , submoduleUser :: !Text
    , submoduleRepo :: !Text
    , submoduleCommit :: !(Maybe Text)
    } deriving (Eq, Ord, Show)

Current implicit behaviour:
{-# SUBMODULE ambiata/p@xxx #-} -> Submodule Ssh "ambiata" "p" "xxx"

Explicit protocol:
{-# SUBMODULE ssh ambiata/p@xxx #-} -> Submodule Ssh "ambiata" "p" "xxx"

{-# SUBMODULE https ambiata/p@xxx #-} -> Submodule Https "ambiata" "p" "xxx"
```

I am sure there is a better way of describing the protocols, just first thing that came to mind that would keep compatibility

Reply:
Yep, that makes sense. Will look into it.

Reply:
Actually, why not make the protocol and the server explicit like:
```
{-# SUBMODULE git@github.com:ambiata/p@xxx #-}
{-# SUBMODULE https://github.com/ambiata/x@xxx #-}
```
so that other services like gitlab and bitbucket can be used.

Reply:
> Actually, why not make the protocol and the server explicit like:

I'm actually 100% on board with this, I never liked hard coding to github. That said, my suggestion is that you should also fall back to the previous behaviour to remain backwards compatible (as well as keeping with Jacob's original intention)